static/js/composebox_typeahead.js

```diff
@@ -855,7 +855,12 @@ exports.content_typeahead_selected = function (item, event) {
         // Stream + topic mention typeahead; close the stream+topic mention syntax
         // with the topic and the final **.
         const start = -this.token.length;
-        beginning = beginning.slice(0, start) + item + "** ";
+        if (start !== 0) {
+            // This is the only case where we can set
+            // up the typeahead for an empty token.
```

Comment:
We should mention that `slice(0,0)` produces an empty string which is why we need this conditional. That would be better.